packages/synapse-react-client/src/components/EntityDownloadButton/EntityDownloadButton.tsx

```diff
@@ -265,6 +304,33 @@ export function EntityDownloadButton(props: {
     props.entityType,
   )
 
+  // Get context and download functionality
+  const { downloadCartPageUrl } = useSynapseContext()
+  const { mutate: addFileToDownloadList } = useAddFileToDownloadList({
+    onSuccess: () => {
+      displayFilesWereAddedToDownloadListSuccess(downloadCartPageUrl)
+    },
```

Comment:
Probably also need to check the `numberOfFilesAdded` here since it could be an empty dataset